apps/charterafrica/src/lib/data/common/index.js

```diff
@@ -171,48 +171,47 @@ export async function processPageFellowships({ blocks }) {
   });
 }
 
-export async function processPageNews({ blocks }) {
-  // TODO(kilemensi): Pull data from CMS
-  blocks.push({
-    slug: "featured-post",
-    category: "News",
-    title: "News Story title goes here and spans over second line",
-    excerpt:
-      "Lorem ipsum dolor sit amet consectetur adipiscing elit tempus nibh cursus, urna porta sagittis non eget taciti nunc sed felis dui, praesent ullamcorper facilisi euismod ut in platea laoreet integer. Lorem ipsum dolor sit amet consectetur",
-    date: "2020-10-10 10:10:10",
-    author: "Author Name",
-    image: {
-      url: "/images/featured_post.jpg",
-      alt: "Featured Post",
-    },
+export async function processPageNews(page, api) {
+  const { blocks, breadcrumbs = [] } = page;
+  const { docs } = await api.getCollection("news", {
+    where: { _status: { equals: "published" } },
+  });
+
+  const processArticle = (data) => ({
+    ...data,
+    author: data?.authors?.map(({ fullName }) => fullName).join(", ") ?? null,
+    image: data?.coverImage ?? null,
+    date: new Date(data?.publishedOn).toUTCString(),
     link: {
-      href: "/knowledge/news",
+      href: `${breadcrumbs[breadcrumbs.length - 1]?.url}/${data?.id}`,
     },
   });
-  blocks.push({
+
+  const articles = docs?.map(processArticle);
+  const rawArticle =
+    blocks.find(({ slug }) => slug === "featured-post")?.featuredPost?.value ??
+    null;
+
+  if (!rawArticle) {
+    return;
+  }
+
+  const featuredNewsPost = processArticle(rawArticle);
+
+  const featuredPost = {
+    category: "News",
+    ...featuredNewsPost,
+    slug: "featured-post",
+  };
+
+  const news = {
     slug: "news",
     title: "News",
-    articles: Array.from({ length: 30 }, (_, i) => ({
-      id: i,
-      title: "News story title goes here and spans over second line. "
-        .repeat((i % 2) + 1)
-        .trim(),
-      date: "2023-02-11",
-      image: {
-        id: "63d2622aafe25f6469605eae",
-        alt: `News Story ${i}`,
-        prefix: "media",
-        filename: `knowledge_${(i % 3) + 1}.jpg`,
-        mimeType: "image/jpg",
-        filesize: 257010,
-        width: 1236,
-        height: 696,
-        createdAt: "2023-01-26T11:21:14.868Z",
-        updatedAt: "2023-01-26T11:21:14.868Z",
-        url: `/images/knowledge_${(i % 3) + 1}.jpg`,
-      },
-    })),
-  });
+    articles,
+  };
+
+  blocks[0] = featuredPost;
+  blocks[1] = news;
```

Comment:
1. featured-post block should already be on page.
2. We can just `push` news block into blocks i.e. it should go last regardless of whether we have featured-post or not.